source/rrStringUtils.h

```diff
@@ -103,5 +107,250 @@ RR_DECLSPEC string              substitute(const string& src, const string& this
 RR_DECLSPEC string              substitute(const string& src, const string& thisOne, const int& withThisOne, const int& howMany = -1);
 //RR_DECLSPEC string              substitute(const string& src, const string& thisOne, const double& withThisOne, const int& howMany = -1);
 RR_DECLSPEC string              removeNewLines(const string& str, const int& howMany = -1);
+
+
+
+template <typename T>
+inline void                saveBinary(std::ostream& out, const T& t)
```

Comment:
Why write the definition in the header file rather than cpp file?

Reply:
Should we also add RR_DECLSPEC to these new functions?
See https://docs.microsoft.com/en-us/cpp/build/exporting-from-a-dll-using-declspec-dllexport?view=vs-2019